### PR TITLE
feat: add ability to parse dev dependencies

### PR DESCRIPTION
**Pull request by gitphill**
- [X] Ready for review
- [X] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Pass optional dev option to parser.
Refactored tests moving fixtures into their own folder.
Added @snyk/cli-interface dependecy and extended BaseInspectOptions (to
get dev option).
Added test for dev option.

#### What are the relevant tickets?

BST-917

### CODE REVIEW

**Comment by gitphill** [comment]
should we have package-lock.json in this repo?

**Comment by lili2311** [comment]
@gitphill not in public packages no :)

File: test/fixtures/circular_deps_php_project/composer_deps.json [hunk 1]
```diff
@@ -268,14 +395,29 @@
...
                             "name": "php",
                             "version": "7.1",
-                            "dependencies": {}
+                            "dependencies": {},
+                            "labels": {
+                              "scope": "prod"
+                            }
                           }
+                        },
+                        "labels": {
```
Comment (lili2311) [inline test/fixtures/circular_deps_php_project/composer_deps.json]: this looks weird, what are these emoty labels? Labels usually accompany a dep entry

Comment (gitphill) [reply test/fixtures/circular_deps_php_project/composer_deps.json]: they have scope 'dev' or 'prod' to indicate whether the dependency is part of dev tree (require-dev) or prod tree (require).

is diff misleading? this label is on the "doctrine/instantiator" dependency and has scope "prod".

**Pull request merged**